### PR TITLE
Add python More Builtin types documentation

### PR DESCRIPTION
**Pull request by koustuvsinha**
Converting python tutorial docs from https://github.com/alayek/python3notes from ipynb to markdown, by preserving formatting. Continuing work from [this PR](https://github.com/alayek/python3notes/pull/1)

### CODE REVIEW

**Comment by alayek** [comment]
Thanks again @koustuvsinha! Other than the TODOs, everything else looks fine.

@Rafase282 @hallaathrad @raisedadead kindly take a look.

File: python-more-builtin-types.md [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+Some objects contain references to other objects; these are called containers.
+Examples of containers are tuples, lists and dictionaries. The references are
+part of a container’s value. In most cases, when we talk about the value of a
+container, we imply the values, not the identities of the contained objects;
+however, when we talk about the mutability of a container, only the identities
+of the immediately contained objects are implied. So, if an immutable container
+(like a tuple) contains a reference to a mutable object, its value changes if
+that mutable object is changed.
+
+https://docs.python.org/3/reference/datamodel.html#objects-values-and-types
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here. `[ title ] ( url )` without spaces.

File: python-more-builtin-types.md [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+
+## Iterators
+
+Python supports a concept of iteration over containers. This is implemented
+using two distinct methods; these are used to allow user-defined classes to
+support iteration.
+
+https://docs.python.org/3/library/stdtypes.html#iterator-types
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here.

File: python-more-builtin-types.md [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+```
+
+## Sequence Types
+
+These represent finite ordered sets indexed by non-negative numbers. The
+built-in function `len()` returns the number of items of a sequence. When the
+length of a sequence is `n`, the index set contains the numbers `0, 1, ...,
+n-1`. Item `i` of sequence a is selected by `a[i]`.
+
+https://docs.python.org/3/reference/datamodel.html#the-standard-type-hierarchy
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here.

File: python-more-builtin-types.md [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+
+* Built-in sequence types are:
+    * `list`
+    * `tuple`
+    * `range`
+* Built-in sequence types are iterable types (implement required `__iter__()`
+method).
+* Operations:
+    * https://docs.python.org/3/library/stdtypes.html#common-sequence-operations
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here.

File: python-more-builtin-types.md [hunk 5]
```diff
@@ -0,0 +1,582 @@
...
+    * https://docs.python.org/3/library/stdtypes.html#mutable-sequence-types
+    * https://docs.python.org/3/library/stdtypes.html#immutable-sequence-types
+
+### Lists
+
+**TODO: `list` basic info**
+
+https://docs.python.org/3/library/stdtypes.html#lists
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here. `[ title ] ( url )` without spaces.

File: python-more-builtin-types.md [hunk 6]
```diff
@@ -0,0 +1,582 @@
...
+>>> heterogeneous_list = [1, "Hello Campers!"]
+>>> print(heterogeneous_list)
+[1, "Hello Campers!"]
+>>> len(heterogeneous_list)
+2
+```
+
+The `list` constructor can also be used to create a `list`:
+
+https://docs.python.org/3/library/stdtypes.html#list
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here.

File: python-more-builtin-types.md [hunk 7]
```diff
@@ -0,0 +1,582 @@
...
+```
+
+**Mutable:**
+
+`lists` are mutable containers. Mutable containers are containers that allow
+changes to which objects are contained by the container. **TODO: ADD MORE?**
+
+**TODO: Which of these should be discussed here:**
+
+https://docs.python.org/3/tutorial/datastructures.html#more-on-lists
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here.

File: python-more-builtin-types.md [hunk 8]
```diff
@@ -0,0 +1,582 @@
...
+Reverse the elements of the list in place.
+
+* `list.copy()`
+Return a shallow copy of the list. Equivalent to a[:].
+
+### Tuples
+
+**TODO: `Tuple` basic info**
+
+https://docs.python.org/3/library/stdtypes.html#tuples
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here.

File: python-more-builtin-types.md [hunk 9]
```diff
@@ -0,0 +1,582 @@
...
+(1, 2, 3, 4)
+>>> 1, 2, 3 == (1, 2, 3)     # Equivalent to 1, 2, (3 == (1, 2, 3))
+(1, 2, False)
+>>> (1, 2, 3) == (1, 2, 3)   # Use surrounding round brackets when ambiguous.
+True
+```
+
+A `tuple` can also be created with the `tuple` constructor:
+
+https://docs.python.org/3/library/stdtypes.html#tuple
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here.

File: python-more-builtin-types.md [hunk 10]
```diff
@@ -0,0 +1,582 @@
...
+>>> greeting = ["Hello", "campers!"]
+>>> enumerator = enumerate(greeting)
+>>> enumerator.next()
+>>> enumerator.__next__()
+(0, 'Hello')
+>>> enumerator.__next__()
+(1, 'campers!')
+```
+
+https://docs.python.org/3/tutorial/datastructures.html#tuples-and-sequences
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here.

File: python-more-builtin-types.md [hunk 11]
```diff
@@ -0,0 +1,582 @@
...
+
+### Range
+
+**TODO: `range` basic info**
+
+https://docs.python.org/3/library/stdtypes.html#ranges
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here.

File: python-more-builtin-types.md [hunk 12]
```diff
@@ -0,0 +1,582 @@
...
+```
+
+**Benefits:**
+
+The benefit of using `range` is that regardless of how large of a range
+specified, only a small amount of memory is needed to store the `range`, the
+values for start, stop, and step. The individual values of the `ranges` are
+calculated upon iteration.
+
+https://docs.python.org/3/library/sys.html#sys.getsizeof
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here.

File: python-more-builtin-types.md [hunk 13]
```diff
@@ -0,0 +1,582 @@
...
+## Big O?
+
+**TODO: Find external big-O resource?**
+
+## Mapping Types
+
+> A mapping object maps hashable values to arbitrary objects. Mappings are
+mutable objects.
+
+> https://docs.python.org/3/library/stdtypes.html#mapping-types-dict
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here.

File: python-more-builtin-types.md [hunk 14]
```diff
@@ -0,0 +1,582 @@
...
+that is, values containing lists, dictionaries or other mutable types (that are
+compared by value rather than by object identity) may not be used as keys.
+Numeric types used for keys obey the normal rules for numeric comparison: if two
+numbers compare equal (such as `1` and `1.0`) then they can be used
+interchangeably to index the same dictionary entry. (Note however, that since
+computers store floating-point numbers as approximations it is usually unwise to
+use them as dictionary keys.)
+
+**TODO: See if anything from
+https://docs.python.org/3/tutorial/datastructures.html#dictionaries should be
```
Comment (Rafase282) [inline python-more-builtin-types.md]: Please use markdown link here. same as the other one.

**Comment by Rafase282** [comment]
Please change title to `Python-More-Builtin-Types.md`

**Pull request merged**